src/main/java/org/jmt/mcmt/serdes/filter/AutoFilter.java

```diff
@@ -0,0 +1,48 @@
+package org.jmt.mcmt.serdes.filter;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.jmt.mcmt.serdes.ISerDesHookType;
+import org.jmt.mcmt.serdes.SerDesRegistry;
+import org.jmt.mcmt.serdes.pools.ChunkLockPool;
+import org.jmt.mcmt.serdes.pools.ISerDesPool;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class AutoFilter implements ISerDesFilter {
+	private static AutoFilter SINGLETON;
+	
+	private ISerDesPool pool;
+	private Set<Class<?>> blacklist = ConcurrentHashMap.newKeySet();
+	
+	public static AutoFilter singleton() {
+		if (SINGLETON == null) SINGLETON = new AutoFilter();
+		return SINGLETON;
+	}
+	
+	@Override
+	public void init() {
+		pool = SerDesRegistry.getOrCreatePool("AUTO", ChunkLockPool::new);
+	}
+	
+	@Override
+	public void serialise(Runnable task, Object obj, BlockPos bp, World w, ISerDesHookType hookType) {
+		pool.serialise(task, obj, bp, w, null);
+	}
+
+	@Override
+	public Set<Class<?>> getTargets() {
+		return blacklist;
+	}
+
+	@Override
+	public ClassMode getModeOnline(Class<?> c) {
+		return ClassMode.BLACKLIST;
```

Comment:
This will cause every unknown class to be blacklisted and run through this filter, which may not be intended behaviour, and definately isn't really desirable; this should return `ClassMode.UNKNOWN` as it's not got any of it's own filtering logic.